mlir/lib/Dialect/Affine/Utils/Utils.cpp

```diff
@@ -178,8 +181,9 @@ class AffineApplyExpander
     Value quotient = builder.create<arith::DivSIOp>(loc, dividend, rhs);
     Value negatedQuotient =
         builder.create<arith::SubIOp>(loc, zeroCst, quotient);
-    Value incrementedQuotient =
-        builder.create<arith::AddIOp>(loc, quotient, oneCst);
+    Value incrementedQuotient = builder.create<arith::AddIOp>(
```

Comment:
I don't have the detailed proof on me, but I suspect that this is going to fail on `quotient = -1`